src/Symfony/Component/HttpKernel/Controller/ArgumentResolver.php

```diff
@@ -37,15 +38,23 @@ final class ArgumentResolver implements ArgumentResolverInterface
     public function __construct(ArgumentMetadataFactoryInterface $argumentMetadataFactory = null, iterable $argumentValueResolvers = [])
     {
         $this->argumentMetadataFactory = $argumentMetadataFactory ?? new ArgumentMetadataFactory();
-        $this->argumentValueResolvers = $argumentValueResolvers ?: self::getDefaultArgumentValueResolvers();
+        $this->argumentValueResolvers = $argumentValueResolvers ?: iterator_to_array(self::getIndexedDefaultArgumentValueResolvers());
     }
 
     public function getArguments(Request $request, callable $controller, \ReflectionFunctionAbstract $reflector = null): array
     {
         $arguments = [];
 
         foreach ($this->argumentMetadataFactory->createArgumentMetadata($controller, $reflector) as $metadata) {
-            foreach ($this->argumentValueResolvers as $resolver) {
+            if ($attributes = $metadata->getAttributesOfType(ValueResolver::class)) {
+                $resolverId = reset($attributes)->id;
```

Comment:
$resolverId leaks from one iteration to another, isn't it?
+ no need to use `reset()`, `[0]` would work just great